os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_common/ameba_i2c.c

```diff
@@ -717,6 +718,14 @@ u32 I2C_MasterRead(I2C_TypeDef *I2Cx, u8 *pBuf, u32 len)
 				I2C_ClearAllINT(I2Cx);
 				return cnt;
 			}
+			DelayUs(1);
+			if (timeout == 0) {
+				RTK_LOGW(TAG, "I2C_MasterRead timeout - resetting I2C\n");
```

Comment:
could you let me know which Tizen Lite api is called by RTK_LOGW?

Reply:
DiagVSprintf

Reply:
do you have any suggestion which API to use?

Reply:
As far as I know, Realtek has other API to flash logs into SW buffer, not into uart hw. I don't know it. Could you check it in Realtek internally?

Reply:
I see, let me check this, but i heard from Eunwoo that printf/dbg should not be used as it will be used in idle_pm and will cause context switching..

Reply:
How about remove log and return error?

Reply:
I think it would be better. let me change it